batch/batch/front_end/query.py

```diff
@@ -0,0 +1,591 @@
+        assert result
+        operator, _, inst_coll = result
+        self.inst_coll = inst_coll
+        self.operator = operator
+
+    def query(self) -> Tuple[str, List[Any]]:
+        op = self.operator.to_sql()
+        if isinstance(self.operator, PartialMatchOperator):
+            self.inst_coll = f'%{self.inst_coll}%'
+        sql = f'(jobs.inst_coll {op} %s)'
+        return (sql, [self.inst_coll])
+
+
+class QuotedExactMatchQuery(Query):
+    @staticmethod
+    def is_valid_query(term: str) -> bool:
+        if len(term) < 3:
+            return False
+        if term[0] != '"':
+            return False
+        return term[-1] == '"'
+
+    def __init__(self, term: str):
+        assert self.is_valid_query(term)
+        self.term = term.lstrip('"').rstrip('"')
+
+    def query(self) -> Tuple[str, List[Any]]:
+        sql = '''
+(((jobs.batch_id, jobs.job_id) IN
+ (SELECT batch_id, job_id FROM job_attributes
+  WHERE `key` = %s OR `value` = %s)) OR
+((jobs.batch_id, jobs.job_id) IN
+ (SELECT batch_id, job_id FROM attempts
+  WHERE instance_name = %s)))
+'''
+        return (sql, [self.term, self.term, self.term])
+
+
+class UnquotedPartialMatchQuery(Query):
+    @staticmethod
+    def is_valid_query(term: str) -> bool:
+        if len(term) < 3:
+            return False
+        if term[0] == '"':
+            return False
+        return term[-1] != '"'
+
+    def __init__(self, term: str):
+        assert self.is_valid_query(term)
+        self.term = term
+
+    def query(self) -> Tuple[str, List[Any]]:
+        sql = '''
+(((jobs.batch_id, jobs.job_id) IN
+ (SELECT batch_id, job_id FROM job_attributes
+  WHERE `key` LIKE %s OR `value` LIKE %s)) OR
+((jobs.batch_id, jobs.job_id) IN
+ (SELECT batch_id, job_id FROM attempts
+  WHERE instance_name LIKE %s)))
+'''
+        escaped_term = f'%{self.term}%'
+        return (sql, [escaped_term, escaped_term, escaped_term])
+
+
+class KeyWordQuery(Query):
+    @staticmethod
+    def is_valid_query(term: str) -> bool:
+        search_result = parse_search_by_operator_type(MatchOperator, None, term)
+        if search_result is None:
+            return False
+        return True
+
+    def __init__(self, term: str):
+        result = parse_search_by_operator_type(MatchOperator, None, term)
+        assert result
+        operator, key, value = result
+        self.key = key
+        self.value = value
+        self.operator = operator
+
+    def query(self) -> Tuple[str, List[Any]]:
+        op = self.operator.to_sql()
+        if isinstance(self.operator, PartialMatchOperator):
+            self.value = f'%{self.value}%'
+        sql = f'''
+((jobs.batch_id, jobs.job_id) IN
+ (SELECT batch_id, job_id FROM job_attributes
+  WHERE `key` = %s AND `value` {op} %s))
+        '''
+        return (sql, [self.key, self.value])
+
+
+class TimeQuery(Query, abc.ABC):
+    @classmethod
+    def from_data(cls, field: str, operator: Operator, time_msecs: int):
+        q = object.__new__(cls)
```

Comment:
Using `object.__new__(cls)` is a smell to me. It looks to me like this method actually better suits the needs of a constructor, since the fields are "prepared" so to speak. The actual constructor needs to do work on its inputs in order to obtain values for fields, so the current constructor seems like it should be some static method that does those things and then calls the constructor with the prepared fields.